test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestNotCompilable.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package ir_framework.tests;
+
+import compiler.lib.ir_framework.*;
+import compiler.lib.ir_framework.shared.TestRunException;
+import compiler.lib.ir_framework.driver.TestVMException;
+import compiler.lib.ir_framework.driver.irmatching.IRViolationException;
+
+/*
+ * @test
+ * @requires vm.compiler2.enabled & vm.flagless
```

Comment:
Hi, I wonder if we should add a `vm.debug == true` to the @requires list? I witnessed test failure on my linux-aarch64 platform with release build when running `make test TEST="testlibrary_tests/ir_framework/tests/TestNotCompilable.java"`

Reply:
@RealFYang Thanks for the report! I ran full testing, but somehow this slipped though!

```
java.lang.RuntimeException: should have thrown TestRunException/TestVMException or IRViolationException
	at ir_framework.tests.TestNotCompilable.runWithExcludeExpectFailure(TestNotCompilable.java:100)
	at ir_framework.tests.TestNotCompilable.runTests(TestNotCompilable.java:58)
	at ir_framework.tests.TestNotCompilable.main(TestNotCompilable.java:41)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at com.sun.javatest.regtest.agent.MainActionHelper$AgentVMRunnable.run(MainActionHelper.java:335)
	at java.base/java.lang.Thread.run(Thread.java:1447)
```

Reply:
I'll file a bug for it!

Reply:
Filed: [JDK-8352597](https://bugs.openjdk.org/browse/JDK-8352597)
[IR Framework] test bug: TestNotCompilable.java fails on product build